ESACCI/read-biomass-ESACCI.py

```diff
@@ -0,0 +1,255 @@
+                    ((np.arange(nyears)*365)[:,np.newaxis]+month_bnd[+1:]).flatten()]).T
+tbnd0 += (start_yr-1850)*365
+tbnd0.shape
+
+tbnd   = np.ma.masked_array(np.random.rand(1,2))
+
+tbnd[0,0] = tbnd0[0,0]
+tbnd[0,1] = tbnd0[nyears-1,1]
+
+t     = tbnd.mean(axis=1)
+t.shape
+
+# Create new spatial dimension
+res    = 0.5
+latbnd = np.asarray([np.arange(- 90    , 90     ,res),
+                     np.arange(- 90+res, 90+0.01,res)]).T
+lonbnd = np.asarray([np.arange(-180    ,180     ,res),
+                     np.arange(-180+res,180+0.01,res)]).T
+lat    = latbnd.mean(axis=1)
+lon    = lonbnd.mean(axis=1)
+
+# Create some fake data
+data   = np.ma.masked_array(np.random.rand(t.size,lat.size,lon.size))
+area   = np.ma.masked_array(np.random.rand(lat.size,lon.size))
+
+nlat = lat.size
+nlon = lon.size
+
+R    = np.float64(6371007.181)
+
+for ny in range(nlat):
+    if ny == 0:
+       dlat = abs((lat[1]-lat[0])*0.01745)
+    else:
+       dlat = abs((lat[ny]-lat[ny-1])*0.01745)
+    dy = R*dlat
+    for nx in range(nlon):
+        if nx==0:
+           dlon = abs((lon[1]-lon[0])*0.01745)
+        else:
+           dlon = abs((lon[nx]-lon[nx-1])*0.01745)
+        dx = R*math.cos(0.01745*lat[ny])*dlon
+        area[ny,nx] = dx*dy
+
+def rebin(arr, new_shape):
+    """Rebin 2D array arr to shape new_shape by averaging."""
+    shape = (new_shape[0], arr.shape[0] // new_shape[0],
+             new_shape[1], arr.shape[1] // new_shape[1])
+    return arr.reshape(shape).mean(-1).mean(1)
+
+# read single netCDF file
+filename = DataDir + '/' + sourceID + '/' + local_source
+print(filename)
+ag=Dataset(filename,'r',format='NETCDF4')
+print(ag.variables) 
+
+#print(ba.variables.keys())
+agb1 = ag.variables['agb']
+lat1 = ag.variables['lat']
+lon1 = ag.variables['lon']
+
+print(agb1)
+
+agb  = agb1[0,::-1,:]
+lat1 = lat1[::-1]
+
+del agb1
+
+nlat1 = lat1.size
+nlon1 = lon1.size
+
+print(nlat1)
+print(nlon1)
+
+nlat2 = int(nlat1/2)
+nlon2 = int(nlon1/2)
+
+print(nlat2)
+print(nlon2)
+
+print(lat1[0:nlat2])
+print(lon1[0:nlon2])
+
+#abgb = np.short(agb[0:nlat2,0:nlon2], dtype=np.float64, order='C')
+
+biomass = np.ma.masked_array(np.random.rand(nlat,nlon), dtype=float64)
+
+biomass[...] = 0.
+
+for ny in range(nlat1):
+    if ny == 0:
+       dlat = abs((lat1[1]-lat1[0])*0.01745)
+    else:
+       dlat = abs((lat1[ny]-lat1[ny-1])*0.01745)
+    dy = R*dlat
+    for nx in range(nlon1):
+        if nx==0:
+           dlon = abs((lon1[1]-lon1[0])*0.01745)
+        else:
+           dlon = abs((lon1[nx]-lon1[nx-1])*0.01745)
+        dx = R*math.cos(0.01745*lat1[ny])*dlon
+        areas = dx*dy*1.0e-4
+
+        #+++++ to generate data in 0.5x0.5 ++++
+        iy   = int(lat1[ny]*2+90.*2)
+        ix   = int(lon1[nx]*2+180.*2)
+
+        # obtain total biomass mass at each grid cell
+        biomass[iy,ix] = biomass[iy,ix] + agb[ny,nx]*areas
+
+#biomass = rebin(abgb, (int(nlat/2), int(nlon/2)))
+
+print('Total biomass')
```

Comment:
print statements are useful for debugging purposes while encoding the data, but not as nice once you have the script working. Please remove the print statements throughout the code :)